evcouplings/compare/hmmer_wrappers.py

```diff
@@ -0,0 +1,309 @@
+        cmd += [
+            "-T", str(seq_threshold),
+            "--domT", str(domain_threshold),
+            "--incT", str(seq_threshold),
+            "--incdomT", str(domain_threshold)
+        ]
+    else:
+        cmd += [
+            "-E", str(seq_threshold),
+            "--domE", str(domain_threshold),
+            "--incE", str(seq_threshold),
+            "--incdomE", str(domain_threshold)
+        ]
+
+    # number of CPUs
+    if cpu is not None:
+        cmd += ["--cpu", str(cpu)]
+
+    # bias correction filter
+    if nobias:
+        cmd += ["--nobias"]
+
+    cmd += [hmmfile, database]
+
+    return_code, stdout, stderr = run(cmd)
+
+    # also check we actually created some sort of alignment
+    verify_resources(
+        "hmmsearch returned empty alignment: "
+        "stdout={} stderr={} file={}".format(
+            stdout, stderr, result.alignment
+        ),
+        result.alignment
+    )
+
+    return result
+
+
+
+def hmmbuild_and_search(database, prefix,
+                  use_bitscores, domain_threshold, seq_threshold,
+                  nobias=False, cpu=None,
+                  stdout_redirect=None,
+                  binary=["hmmbuild", "hmmsearch"]):
+    """
+    Search profile(s) against a sequence database.
+    Refer to HMMER documentation for details.
+
+    http://eddylab.org/software/hmmer3/3.1b2/Userguide.pdf
+
+    Parameters
+    ----------
+    database : str
+        File containing sequence database
+    prefix : str
+        Prefix path for output files. Folder structure in
+        the prefix will be created if not existing.
+    use_bitscores : bool
+        Use bitscore inclusion thresholds rather than E-values.
+
+   domain_threshold : int or float or str
+        Inclusion threshold applied on the domain level
+        (e.g. "1E-03" or 0.001 or 50)
+    seq_threshold : int or float or str
+        Inclusion threshold applied on the sequence level
+        (e.g. "1E-03" or 0.001 or 50)
+    nobias : bool, optional (default: False)
+        Turn of bias correction
+    cpu : int, optional (default: None)
+        Number of CPUs to use for search. Uses all if None.
+    stdout_redirect : str, optional (default: None)
+        Redirect bulky stdout instead of storing
+        with rest of results (use "/dev/null" to dispose)
+    binary : str (default: ["hmmbuild", "hmmsearch"])
+        Path to jackhmmer binary (put in PATH for
+        default to work)
+
+    Returns
+    -------
+    HmmsearchResult
+        namedtuple with fields corresponding to the different
+        output files (prefix, alignment, output, tblout, domtblout)
+
+    Raises
+    ------
+    ExternalToolError, ResourceError
+    """
+    verify_resources(
+        "Input file does not exist or is empty",
+        database
+    )
+
+    # name of the protein (prefix)
+    protein_name = path.basename(prefix)
+
+    # path to the main directory
+    main_dir = path.normpath(path.dirname(prefix) + '/../..')
```

Comment:
Same here... what is outside or above the prefix is no business of the protocol. The outside code use a  different directory tree structure at some point (which I have done before), and this would break it.